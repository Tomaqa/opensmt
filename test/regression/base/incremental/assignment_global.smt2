(set-option :global-declarations true)
(set-option :produce-assignments true)
(set-logic QF_UF)
(declare-fun a () Bool)
(declare-fun b () Bool)
(assert (or a b))
(push 1)
(assert (! a :named n0))
(check-sat)
(pop 1)
(assert (! (xor a b) :named n1))
(assert (! a :named n2))
(check-sat)
(get-assignment)
; NOTE: Because of global declarations, the name n0 (and variable a) are still valid even after pop
