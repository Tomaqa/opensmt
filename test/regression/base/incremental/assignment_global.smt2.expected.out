sat
sat
((n0 true) (n1 unknown) (n2 true))
